internal/cmd/shell.go

```diff
@@ -188,13 +189,26 @@ func runCommand(ctx context.Context, envName, command string, args []string) err
 	if err := process.Wait(); err != nil {
 		var closeErr websocket.CloseError
 		if xerrors.Is(err, ctx.Err()) || xerrors.As(err, &closeErr) {
-			return xerrors.Errorf("network error, is %q online?", envName)
+			return networkErr(client, env)
 		}
 		return err
 	}
 	return nil
 }
 
+func networkErr(client *coder.Client, env *coder.Environment) error {
+	if env.LatestStat.ContainerStatus != coder.EnvironmentOn {
+		return clog.Fatal(
+			"environment is not running",
+			fmt.Sprintf("environment %q is not running", env.Name),
+			fmt.Sprintf("its current status is %q", env.LatestStat.ContainerStatus),
+			"",
```

Comment:
The empty string on a lot of these are a bit confusing, I know it's a bit redundant but maybe something like `clog.LineSeparator` or similar would make it clear why it's used.

Reply:
Oh nice nice.